File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/service/SampleWorker3.java
package com.jdroid.android.sample.ui.service;

import android.content.Context;

import com.jdroid.android.jetpack.work.AbstractWorker;
import com.jdroid.java.concurrent.ExecutorUtils;

import java.util.concurrent.TimeUnit;

import androidx.annotation.NonNull;
import androidx.work.Data;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class SampleWorker3 extends AbstractWorker {

	public SampleWorker3(@NonNull Context context, @NonNull WorkerParameters workerParams) {
		super(context, workerParams);
	}

	@NonNull
	@Override
	protected Worker.Result onWork() {
		ExecutorUtils.sleep(30, TimeUnit.SECONDS);
		Data.Builder outputBuilder = new Data.Builder();
		outputBuilder.putString("result", "ok");
		setOutputData(outputBuilder.build());
		return Result.SUCCESS;
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/service/SampleWorker3.kt
package com.jdroid.android.sample.ui.service

import android.content.Context
import androidx.work.Data
import androidx.work.WorkerParameters
import androidx.work.workDataOf
import com.jdroid.android.jetpack.work.AbstractWorker
import com.jdroid.java.concurrent.ExecutorUtils.sleep
import java.util.concurrent.TimeUnit

class SampleWorker3(context: Context, workerParams: WorkerParameters) : AbstractWorker(context, workerParams) {
    override fun onWork(): Result {
        sleep(30, TimeUnit.SECONDS)
        val outputData = workDataOf("result" to "ok")
        return Result.success(outputData)
    }
}